README.md

```diff
@@ -48,15 +47,34 @@ down to the following steps:
 Check out our series of [example projects](./examples) which showcase how to use
 `cargo2nix` in detail.
 
+### As a build system (experimental Nix Flakes)
+
+The basic process of converting an existing Cargo project to `cargo2nix` boils
+down to the following steps:
+
+1. Generate a `Cargo.nix` file by running `cargo2nix -f` at the root of your
+   Cargo workspace.
+2. Create a `flake.nix` file with nixpkgs, [cargo2nix] and [nixpkgs-mozilla] inputs
+   imports nixpkgs with nixpkgs-mozilla and cargo2nix overlays and builds your project using the `Cargo.nix` file
+   from earlier. You can use [rust-overlay] alternative to [nixpkgs-mozilla] which contains pre-fetched toolchain
+   hashes.
+3. Run `nix build` to compile and/or test your project.
+
+[nixpkgs-mozilla]: https://github.com/mozilla/nixpkgs-mozilla#rust-overlay
+[rust-overlay]: https://github.com/oxalica/rust-overlay
+[cargo2nix]: ./flake.nix
+
+Check out our [example flake](./examples-flakes/1-hello-world) which showcases how to use
+`cargo2nix` flake in detail.
+
 ### Declarative debug & development shell
 
 You can load a `nix-shell` for any crate derivation in the dependency tree. The
 advantage of this shell is that in this environment users can develop their
 crates and be sure that their crates builds in the same way that `cargo2nix`
 overlay will build them.
 
-To do this, first run `nix-shell -A 'rustPkgs.<registry>.<crate>."x.y.z"'
-default.nix`.  For instance, the following command being invoked in this
+To do this, first run `nix-shell -A 'rustPkgs.<registry>.<crate>."x.y.z"' default.nix`. For instance, the following command being invoked in this
```

Comment:
Is there a `nix dev` equivalent to also mention?

Reply:
not that I know of